### PR TITLE
define new keyboard "BusinessCard"

### PR DESCRIPTION
**Pull request by kakunpc**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/business_card/alpha/config.h [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+//   #define RGBLIGHT_EFFECT_SNAKE
+//   #define RGBLIGHT_EFFECT_KNIGHT
+//   #define RGBLIGHT_EFFECT_CHRISTMAS
+//   #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (fauxpark) [inline keyboards/business_card/alpha/config.h]:
```suggestion
#define DEBOUNCE 5
```

`DEBOUNCING_DELAY` has been deprecated in favour of `DEBOUNCE`.

File: keyboards/business_card/alpha/config.h [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
```
Comment (fauxpark) [inline keyboards/business_card/alpha/config.h]:
```suggestion
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    SKILL,
+    SOCIAL,
+    ABOUT
+};
+
+const uint16_t PROGMEM
+keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] =
+LAYOUT(/* Base */
+        SKILL, SOCIAL, ABOUT, \
```
Comment (fauxpark) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
        SKILL, SOCIAL, ABOUT,
```

File: keyboards/business_card/alpha/rules.mk [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/business_card/alpha/rules.mk]:
```suggestion
```

File: keyboards/business_card/beta/beta.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+void led_set_kb(uint8_t usb_led) {
+  // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+  led_set_user(usb_led);
+}
+
+*/
+
+void matrix_scan_kb(void)
```
Comment (fauxpark) [inline keyboards/business_card/beta/beta.c]: You can safely remove this function as the default `matrix_scan_kb()` does exactly this.

File: keyboards/business_card/beta/config.h [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+//   #define RGBLIGHT_EFFECT_SNAKE
+//   #define RGBLIGHT_EFFECT_KNIGHT
+//   #define RGBLIGHT_EFFECT_CHRISTMAS
+//   #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (fauxpark) [inline keyboards/business_card/beta/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/business_card/beta/config.h [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
```
Comment (fauxpark) [inline keyboards/business_card/beta/config.h]:
```suggestion
```

File: keyboards/business_card/beta/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+#include <stdio.h>
+#include <math.h>
+#include "rgblight_list.h"
+#include "oled_driver.h"
+
+const uint16_t PROGMEM
+keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] =
+LAYOUT(/* Base */
+        KC_1, KC_2, \
```
Comment (fauxpark) [inline keyboards/business_card/beta/keymaps/default/keymap.c]:
```suggestion
        KC_1, KC_2,
```

File: keyboards/business_card/beta/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        KC_3, KC_4, \
```
Comment (fauxpark) [inline keyboards/business_card/beta/keymaps/default/keymap.c]:
```suggestion
        KC_3, KC_4,
```

File: keyboards/business_card/beta/rules.mk [hunk 10]
```diff
@@ -0,0 +1,82 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/business_card/beta/rules.mk]:
```suggestion
```

File: keyboards/business_card/rules.mk [hunk 11]
```diff
@@ -0,0 +1,87 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/business_card/rules.mk]:
```suggestion
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+};
+
+int type = HOME;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case NAME:
+            if (record->event.pressed) {
+                type = NAME;
+                rgblight_sethsv_blue();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]: Especially if you're doing this often, you DO NOT want to use the EEPROM version. 

```suggestion
                rgblight_sethsv_noeeprom_blue();
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+                SEND_STRING("2");
+            }
+            break;
+        case ADDRESS:
+            if (record->event.pressed) {
+                type = ADDRESS;
+                rgblight_sethsv_red();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
                rgblight_sethsv_noeeprom_red();
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+                SEND_STRING("3");
+            }
+            break;
+        case SKILL:
+            if (record->event.pressed) {
+                type = SKILL;
+                rgblight_sethsv_yellow();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
                rgblight_sethsv_noeeprom_yellow();
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+            if (record->event.pressed) {
+                type = SOCIAL;
+                rgblight_sethsv_cyan();
+                SEND_STRING("5");
+            }
+            break;
+        case ABOUT:
+            if (record->event.pressed) {
+                type = ABOUT;
+                rgblight_sethsv_pink();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
                rgblight_sethsv_noeeprom_pink();
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+                SEND_STRING("6");
+            }
+            break;
+
+        case HOME:
+        default:
+            if (record->event.pressed) {
+                type = HOME;
+                rgblight_sethsv_white();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
                rgblight_sethsv_noeeprom_white();
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    matrix_update(&display, &matrix);
+}
+
+void led_set_user(uint8_t usb_led) {}
+
+void keyboard_post_init_user(void) {
+    rgblight_enable_noeeprom();
+    rgblight_sethsv_white();
```
Comment (drashna) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
    rgblight_sethsv_noeeprom_white();
```

File: keyboards/business_card/alpha/rules.mk [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
+COMMON_VPATH += keyboards/helix
```
Comment (drashna) [inline keyboards/business_card/alpha/rules.mk]: Nope.

Please don't reference to files in another keyboard's folder.  

```suggestion
```

File: keyboards/business_card/rules.mk [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
+COMMON_VPATH += keyboards/helix
```
Comment (drashna) [inline keyboards/business_card/rules.mk]:
```suggestion
```

**Comment by kakunpc** [comment]
Sorry, I forgot to erase.

File: keyboards/business_card/beta/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "rgblight_list.h"
+#include "oled_driver.h"
```
Comment (drashna) [inline keyboards/business_card/beta/keymaps/default/keymap.c]: This shouldn't actually be needed.   And I think the rgblight_list shouldn't be either. 
```suggestion
```

File: keyboards/business_card/alpha/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+
+const uint16_t PROGMEM
+keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] =
+LAYOUT(/* Base */
+        KC_1, KC_2, KC_3, \
```
Comment (fauxpark) [inline keyboards/business_card/alpha/keymaps/default/keymap.c]:
```suggestion
        KC_1, KC_2, KC_3,
```

**Comment by drashna** [comment]
Thanks!

**Pull request merged**